File: src/codility/greedy/MaxNonoverlappingSegments.java
package codility.greedy;

public class MaxNonoverlappingSegments {
    public int solution(int[] A, int[] B) {

        int currentEnd = -1;
        int seg =0;
        for (int i = 0;i<A.length;i++) {
            if (A[i] > currentEnd) {
                currentEnd = B[i];
                seg++;
            }
        }
        return seg;
    }

}

File: src/codility/kotlin/MaxNonoverlappingSegments.kt
package codility.kotlin

class MaxNonoverlappingSegments {

    fun solution(A: IntArray, B: IntArray): Int {
        if (A.isEmpty()) {
            return 0
        }

        var count = 1
        var currentEnd = B[0]

        for (i in 1 until A.size) {
            if (currentEnd < A[i]) {
                count++
                currentEnd = B[i]
            }
        }

        return count
    }
}